src/app-logic/url-handling.js

```diff
@@ -116,7 +116,12 @@ function getPathParts(urlState: UrlState): string[] {
     case 'none':
       return [];
     case 'compare':
-      return ['compare'];
+      // Special handling for CompareHome: we shouldn't append anything but the
+      // dataSource when the user is on the comparison form.
+      if (urlState.profilesToCompare === null) {
+        return ['compare'];
+      }
+      return ['compare', urlState.selectedTab];
```

Comment:
optional question: do you think it would be good to also add the `urlState.profiles` here?

Reply:
nvm. thinking about it, it seems better to keep them as query strings to me.

Reply:
As a bit of history: initially I did include them in /compare (something like `/compare/profile1..profile2/`), but then I decided to support adding more than 2 profiles (in that case we wouldn't have the diff profile of course), and this wouldn't work anymore.